engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/UniqueElementsValidator.java

```diff
@@ -38,7 +41,7 @@
 	 * @return true if the input collection is null or does not contain duplicate elements
 	 */
 	@Override
-	public boolean isValid(Collection<?> collection, ConstraintValidatorContext constraintValidatorContext) {
+	public boolean isValid(Collection collection, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
I had to keep the `Collection<?>` in the last method of this class as, otherwise, the stream generics just go berserk.